### PR TITLE
Add missing WebGL interfaces, and a few gl calls

### PR DESCRIPTION
**Pull request by emilio**
See the commit descriptions for more details. 

Blocked on https://github.com/servo/gleam/pull/24

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6357)

### CODE REVIEW

**Comment by pcwalton** [comment]
r=me, looks good!

**Comment by ecoal95** [comment]
By the way, It's not blocked anymore since the gleam por has ben merged :P

File: components/servo/Cargo.lock [hunk 1]
```diff
@@ -1040,6 +1040,7 @@ dependencies = [
  "fnv 1.0.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "geom 0.1.0 (git+https://github.com/servo/rust-geom)",
  "gfx 0.0.1",
+ "gfx_traits 0.0.1",
```
Comment (jdm) [inline components/servo/Cargo.lock]: Is this change intentional?

Comment (nox) [inline components/servo/Cargo.lock]: Seeing #6370, probably not.

Comment (ecoal95) [inline components/servo/Cargo.lock]: Nope, it wasn't.

I noted it but I supposed it was due to some other pr which forgot to update the Cargo.lock files. I'm reverting it :P

**Comment by ecoal95** [comment]
Needed to update gleam, my fault, should be solved now.

**Comment by nox** [comment]
Please reorder commits, this is going to fail to compile if we bisect in the middle of the PR.

**Comment by ecoal95** [comment]
What do you mean? Moving the gleam update before the rest of them?

**Comment by nox** [comment]
Yes.

**Comment by ecoal95** [comment]
That should do it :)

**Comment by nox** [comment]
@bors-servo: r=pcwalton

Cool, thanks!
